docs-sphinx/user-guide/how-to-use-header-only-layoutbuilder.md

```diff
@@ -0,0 +1,260 @@
+
+1. BuilderOptions.h
+2. GrowableBuffer.h
+3. LayoutBuilder.h
+4. utils.h
+
+Three phases of using Layout Builder
+------------------------------------
+
+1. <b>Constructing a Layout Builder:</b> from variadic templates. (It is implicit template instantiation).
+2. <b>Filling the Layout Builder:</b> while repeatedly walking over the raw pointers within the LayoutBuilder.
+3. <b>Taking the data out to user allocated buffers:</b> Then user can pass them to Python if they want.
+
+Example
+-------
+
+Below is an example for creating RecordArray with NumpyArray and ListOffsetArray as fields.
+
+Type alias can be used for each builder class.
+
+```{code-cell}
+#pragma cling add_include_path("../../src/awkward/_v2/cpp-headers")
+
+#include "awkward/LayoutBuilder.h"
+
+#include <string>
+#include <fstream>
+
+#include "nlohmann/json.hpp"
+```
+
+```{code-cell}
+using UserDefinedMap = std::map<std::size_t, std::string>;
+```
+
+```{code-cell}
+template<class PRIMITIVE>
+using NumpyBuilder = awkward::LayoutBuilder::Numpy<PRIMITIVE>;
+
+template<class PRIMITIVE, class BUILDER>
+using ListOffsetBuilder = awkward::LayoutBuilder::ListOffset<PRIMITIVE, BUILDER>;
+
+template<class... BUILDERS>
+using RecordBuilder = awkward::LayoutBuilder::Record<UserDefinedMap, BUILDERS...>; 
+
+template<std::size_t field_name, class BUILDER>
+using RecordField = awkward::LayoutBuilder::Field<field_name, BUILDER>;
+```
+
+Note, it is not possible to template on `std::string` because this feature comes only from `C++20`. That is why, a user-defined map with enumerated type field ID as keys and field names as value has to provided for passing the field names as template parameters to the Record Builder.
+
+If multiple Record Builders are used in a Builder, then a user-defined `field_map` has to be provided for each Record Builder.
+
+```{code-cell}
+enum Field : std::size_t {one, two};
+
+UserDefinedMap fields_map({
+{Field::one, "one"},
+{Field::two, "two"}});
+```
+
+The Record Builder content is a heterogeneous type container (std::tuple) which can take other Builders as template parameters. The field names are non-type template parameters defined by a user. In this example, a Record Builder with the fields as Numpy and ListOffset is taken.
+
+In the ListOffset Builder, there is an option to use 64-bit signed integers `int64`, 32-bit signed integers `int32` or 32-bit unsigned integers `uint32` as the type for list offsets.
+
+To set the field names, either the user-defined `fields_map` can be passed as a parameter in Builder object of `set_field_names()` method can be used.
+
+```{code-cell}
+RecordBuilder<
+  RecordField<Field::one, NumpyBuilder<double>>,
+  RecordField<Field::two, ListOffsetBuilder<int64_t,
+      NumpyBuilder<int32_t>>>
+> builder(fields_map);
+
+//  builder.set_field_names(fields_map);
+```
```

Comment:
or alternative code fragment with setting the map via function and in addition to this one without commented out line?